proxy/proxy.go

```diff
@@ -107,5 +106,5 @@ func main() {
 	})
 
 	level.Info(logger).Log("msg", "Listening", "address", *listenAddress)
-	log.Fatal(http.ListenAndServe(*listenAddress, nil))
+	level.Error(logger).Log("msg", "Listening failed", "err", http.ListenAndServe(*listenAddress, nil))
```

Comment:
Will the process still die if the listen fails?

Reply:
No, it won't. I figured it wouldn't matter much since it was the last line of code in `main`, but now I realize that my patch doesn't give a non-zero exit code (like `log.Fatal` would), so that's wrong.

I see two options: 1. use pkg "log" for the original `Fatal` semantics or 2. add an explicit os.Exit(1)  to my patch. 

What are your thoughts?

Reply:
Explicit exit would be best I think

Reply:
Fixed

Reply:
You should check for the error

Reply:
How do you mean, what is there to check for? From the docs: `ListenAndServe` always returns a non-nil error.

Reply:
Check for the error, it'll make it more obvious what's going on

Reply:
I think I get it; added the check.